cuda_bindings/tests/test_nvjitlink.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: LicenseRef-NVIDIA-SOFTWARE-LICENSE
+
+import pytest
+import os
+from cuda.bindings import nvjitlink
+
+ptx_code = """
+.version 8.5
+.target sm_90
+.address_size 64
+
+.visible .entry _Z6kernelPi(
+    .param .u64 _Z6kernelPi_param_0
+)
+{
+    .reg .pred  %p<2>;
+    .reg .b32   %r<3>;
+    .reg .b64   %rd<3>;
+
+    ld.param.u64    %rd1, [_Z6kernelPi_param_0];
+    cvta.to.global.u64  %rd2, %rd1;
+    mov.u32     %r1, %tid.x;
+    st.global.u32   [%rd2+0], %r1;
+    ret;
+}
+"""
+
+minimal_kernel = """
+.version 6.4
+.target sm_75
+.address_size 64
+
+.visible .entry _kernel() {
+    ret;
+}
+"""
+
+# Convert PTX code to bytes
+ptx_bytes = ptx_code.encode('utf-8')
+minimal_kernel_bytes = minimal_kernel.encode('utf-8')
+
+
+def test_unrecognized_option_error():
+    with pytest.raises(nvjitlink.nvJitLinkError, match="ERROR_UNRECOGNIZED_OPTION"):
+        nvjitlink.create(1, ["-fictitious_option"])
+
+
+def test_invalid_arch_error():
+    # sm_XX is not a valid architecture
+    with pytest.raises(nvjitlink.nvJitLinkError, match="ERROR_UNRECOGNIZED_OPTION"):
+        nvjitlink.create(1, ["-arch=sm_XX"])
+
+
+def test_create_and_destroy():
+    handle = nvjitlink.create(1, ["-arch=sm_53"])
+    assert handle != 0
+    nvjitlink.destroy(handle)
+
+
+def test_complete_empty():
+    handle = nvjitlink.create(1, ["-arch=sm_90"])
+    nvjitlink.complete(handle)
+    nvjitlink.destroy(handle)
+
+def test_add_data():
+    handle = nvjitlink.create(1, ["-arch=sm_90"])
+    data = ptx_bytes
+    nvjitlink.add_data(handle, nvjitlink.InputType.ANY, data, len(data), "test_data")
+
+
+def test_add_file():
+    handle = nvjitlink.create(1, ["-arch=sm_90"])
+    file_path = "test_file.cubin"

```

Comment:
Maybe this is a bit more robust? (so that we don't worry where we launch the tests)
```suggestion
    file_path = os.path.join(os.path.dirname(__file__), "test_file.cubin")
```
(using the "new" path object from `path lib` is also fine, I am "old-school" 😅)